AndroidManifest.xml

```diff
@@ -38,25 +57,70 @@
+        <activity android:name=".ui.setting.ServerEditionActivity" android:parentActivityName=".ui.setting.SettingActivity">
+            <!-- For API v14 & 15 -->
+            <meta-data android:name="android.support.PARENT_ACTIVITY" android:value=".ui.setting.SettingActivity"/>
+        </activity>
+        
+        <!--
+           Documents
+          -->
+        <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.DocumentActivity" android:parentActivityName=".ui.HomeActivity">
+            <!-- For API v14 & 15 -->
+            <meta-data android:name="android.support.PARENT_ACTIVITY" android:value=".ui.HomeActivity"/>
+        </activity>
         <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.DocumentActionDialog"/>
-        <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.DashboardActivity"/>
+        
+        <!--
+           Gadgets (Apps)
+          -->
+        <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.DashboardActivity" android:parentActivityName=".ui.HomeActivity">
+            <!-- For API v14 & 15 -->
+            <meta-data android:name="android.support.PARENT_ACTIVITY" android:value=".ui.HomeActivity"/>
+        </activity>
         <activity android:configChanges="orientation" android:name=".ui.WebViewActivity"/>
-        <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.social.SocialDetailActivity"/>
+        
+        <!--
+           Social (Activity Stream, Details, Composer, Likers)
+          -->
+        <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.social.SocialTabsActivity" android:parentActivityName=".ui.HomeActivity" android:theme="@style/Theme.PageIndicatorDefaults">
+            <!-- For API v14 & 15 -->
+            <meta-data android:name="android.support.PARENT_ACTIVITY" android:value=".ui.HomeActivity"/>
+        </activity>
+        <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.social.SocialDetailActivity" android:parentActivityName=".ui.social.SocialTabsActivity">
+            <!-- For API v14 & 15 -->
+            <meta-data android:name="android.support.PARENT_ACTIVITY" android:value=".ui.social.SocialTabsActivity"/>
+        </activity>
+        <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.social.LikeListActivity" android:parentActivityName=".ui.social.SocialDetailActivity">
+            <!-- For API v14 & 15 -->
+            <meta-data android:name="android.support.PARENT_ACTIVITY" android:value=".ui.social.SocialDetailActivity"/>
+        </activity>
         <activity android:configChanges="keyboardHidden|orientation" android:name=".ui.social.ComposeMessageActivity"/>
+        <activity android:name=".ui.social.SpaceSelectorActivity" android:parentActivityName=".ui.social.ComposeMessageActivity">
```

Comment:
`SpaceSelectorActivity` is called with `startActivityForResult` from 2 different origins : `ShareActivity` and 'ComposeMessageActivity`. It should not set a parent activity like this as it will return nothing to the calling activity.